test/E2ETest/ApplicationTests.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Tye;
+using Test.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace E2ETest
+{
+    public class ApplicationTests
```

Comment:
It would be really nice to see some unit tests of the dependency matching logic (which services get which dependnecies). I'm not 100% convinced that it's correct.